internal/input/changes.go

```diff
@@ -116,97 +110,205 @@ func closeChannels(accChannel, offChannel, trustChannel chan xdr.LedgerEntry) {
+func StreamChanges(core *ledgerbackend.CaptiveStellarCore, start, end, batchSize uint32, accChannel, offChannel, trustChannel chan ChangeBatch, logger *log.Entry) {
 	if end != 0 {
-		for seq := start; seq <= end; seq++ {
-			sendLedgerChangesToChannels(seq, core, accChannel, offChannel, trustChannel)
+		totalBatches := uint32(math.Ceil(float64(end-start+1) / float64(batchSize)))
+		for currentBatch := uint32(0); currentBatch < totalBatches; currentBatch++ {
+			batchStart := start + currentBatch*batchSize
+			batchEnd := batchStart + batchSize
+			if batchEnd > end+1 {
+				batchEnd = end + 1
+			}
+
+			exportBatch(batchStart, batchEnd, core, accChannel, offChannel, trustChannel, logger)
 		}
 	} else {
-		currentLedger := start
+		batchStart := start
+		batchEnd := batchStart + batchSize
 		for {
-			sendLedgerChangesToChannels(currentLedger, core, accChannel, offChannel, trustChannel)
-			currentLedger++
+			exportBatch(batchStart, batchEnd, core, accChannel, offChannel, trustChannel, logger)
+			batchStart = batchEnd
+			batchEnd = batchStart + batchSize
 		}
 	}
 
 	closeChannels(accChannel, offChannel, trustChannel)
 }
 
 // ReceiveChanges reads in the ledger entries from the provided channels, transforms them, and adds them to the slice with the other transformed entries.
-func ReceiveChanges(accChannel, offChannel, trustChannel chan xdr.LedgerEntry, logger *log.Entry) ([]transform.AccountOutput, []transform.OfferOutput, []transform.TrustlineOutput) {
+func ReceiveChanges(accChannel, offChannel, trustChannel chan ChangeBatch, logger *log.Entry) ([]transform.AccountOutput, []transform.OfferOutput, []transform.TrustlineOutput) {
 	transformedAccounts := make([]transform.AccountOutput, 0)
 	transformedOffers := make([]transform.OfferOutput, 0)
 	transformedTrustlines := make([]transform.TrustlineOutput, 0)
-
+	accBatchRead, offBatchRead, trustBatchRead := false, false, false
 	for {
-
 		select {
-		case entry, ok := <-accChannel:
+		case batch, ok := <-accChannel:
 			if !ok {
 				accChannel = nil
 				break
 			}
 
-			acc, err := transform.TransformAccount(entry)
-			if err != nil {
-				errorMsg := fmt.Sprintf("error transforming account entry last updated at: %d", entry.LastModifiedLedgerSeq)
-				logger.Error(errorMsg, err)
-				break
-			}
+			for _, change := range batch.Changes {
+				// TODO: export deleted changes as well as created/updated entries
+				if change.Post == nil {
+					continue
+				}
 
-			transformedAccounts = append(transformedAccounts, acc)
+				entry := *change.Post
+				acc, err := transform.TransformAccount(entry)
+				if err != nil {
+					errorMsg := fmt.Sprintf("error transforming account entry last updated at: %d", entry.LastModifiedLedgerSeq)
+					logger.Error(errorMsg, err)
+					break
+				}
+
+				transformedAccounts = append(transformedAccounts, acc)
+				accBatchRead = true
```

Comment:
I think you're setting this flag as true after reading a single `Change` from each of the panels, rather than from reading an entire batch. It's possible I'm wrong, but if not, make sure to change this, since it's breaking.